src/core/manager.c

```diff
@@ -1019,6 +1021,53 @@ int manager_new(RuntimeScope runtime_scope, ManagerTestRunFlags test_run_flags,
 
                 if (r < 0 && r != -EEXIST)
                         return r;
+
+                m->executor_fd = open(SYSTEMD_EXECUTOR_BINARY_PATH, O_CLOEXEC|O_PATH);
+                if (m->executor_fd < 0)
+                        return log_warning_errno(errno,
+                                                 "Failed to open executor binary '%s': %m",
+                                                 SYSTEMD_EXECUTOR_BINARY_PATH);
+
+                /* posix_spawn cannot take a FD, so pin the path */
+                m->executor_path = strdup(FORMAT_PROC_FD_PATH(m->executor_fd));
```

Comment:
hmm, tbh, i dont like this very much: given that fds come and go and only make sense inside the same process (and similar, pids come and go) I think it would be much nicer if we'd generate these paths only immediately before using them. Or in other words, make exec_spawn() turn the executor_fd into a path right before using it, instead of doing this already here.

(or go even further, to that in posix_spawn_wrapper(), i.e. make it take both path and fd. And if the fd is given use that, and if not, use the path)